driver-dse/src/main/java/com/datastax/driver/graph/AbstractGraphStatement.java

```diff
@@ -0,0 +1,141 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.graph;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+
+abstract class AbstractGraphStatement<T extends Statement> {
+
+    final Map<String, ByteBuffer> payload;
+    T wrappedStatement;
+    final GraphSession session;
+
+    AbstractGraphStatement(GraphSession session) {
+        this.session = session;
+        this.payload = Maps.newHashMap(session.getDefaultGraphOptions());
+    }
+
+    /* TODO: eventually make more advanced checks on the statement
+     *
+     * As of right now, mandatory fields for DSE Graph are :
+     * - graph-keyspace : Must be set by users
+     * - graph-source : default is "default"
+     * - graph-language : default is "gremlin-groovy"
+     *
+     */
+    static boolean checkStatement(AbstractGraphStatement graphStatement) {
+        return graphStatement.getGraphOptions().containsKey(GraphSession.GRAPH_SOURCE_KEY)
+            && graphStatement.getGraphOptions().containsKey(GraphSession.GRAPH_LANGUAGE_KEY)
+            && graphStatement.getGraphOptions().containsKey(GraphSession.GRAPH_SOURCE_KEY);
```

Comment:
Duplicate key GRAPH_SOURCE_KEY, should probably be GRAPH_KEYPSACE_KEY (which has a typo btw).